cmd/nex/cli.go

```diff
@@ -11,10 +11,12 @@ type Globals struct {
 	GlobalLogger `prefix:"logger." group:"Logger Configuration"`
 	GlobalNats   `prefix:"nats." group:"NATS Configuration"`
 
-	Config    kong.ConfigFlag  `help:"Configuration file to load" placeholder:"./nex.config.json"`
-	Version   kong.VersionFlag `help:"Print version information"`
-	Namespace string           `env:"NEX_NAMESPACE" placeholder:"default" help:"Specifies namespace when running nex commands"`
-	Check     bool             `help:"Print the current configuration"`
+	Config              kong.ConfigFlag  `help:"Configuration file to load" placeholder:"./nex.config.json"`
+	Version             kong.VersionFlag `help:"Print version information"`
+	Namespace           string           `env:"NEX_NAMESPACE" placeholder:"default" help:"Specifies namespace when running nex commands"`
+	Check               bool             `help:"Print the current values of all options without running a command"`
+	DisableUpgradeCheck bool             `env:"NEX_DISABLE_UPGRADE_CHECK" name:"disable-upgrade-check" help:"Disable the upgrade check"`
+	AutoUpgrade         bool             `env:"NEX_AUTO_UPGRADE" name:"auto-upgrade" help:"Automatically upgrade the nex CLI when a new version is available"`
```

Comment:
how do these two fields interact with each other? For example, what happens if I enable auto upgrade and disable the upgrade check? Will I automatically upgrade or will I not check (and therefore not upgrade)?

Reply:
Disable turns off all version checking, including auto updates.
So, in order for the auto upgrade to work, Disable must == false (default)

Reply:
can we add a rejection check that flags it as bad config if both of these are true? I don't know if Kong can do mutually exclusive bools

Reply:
done!